subprojects/docs/src/docs/userguide/dependencyManagement.adoc

```diff
@@ -198,102 +198,149 @@ Let's say you wanted to declare a dependency on the link:https://tomcat.apache.o
 
 ++++
-<sample id="exclude-dependencies" dir="userguide/artifacts/excludesAndClassifiers" title="Excluding transitive dependencies">
-                <sourcefile file="build.gradle" snippet="exclude-dependencies"/>
-            </sample>
+<sample id="exclude-transitive-for-dependency" dir="userguide/dependencies/excludingTransitiveDependenciesForDependency" title="Excluding transitive dependency for a particular dependency declaration">
+    <sourcefile file="build.gradle" snippet="exclude-transitive-dependencies"/>
+</sample>
 ++++
 
-If you define an exclude for a particular configuration, the excluded transitive dependency will be filtered for all dependencies when resolving this configuration or any inheriting configuration. If you want to exclude a transitive dependency from all your configurations you can use the Groovy spread-dot operator to express this in a concise way, as shown in the example. When defining an exclude, you can specify either only the organization or only the module name or both. Also look at the API documentation of the api:org.gradle.api.artifacts.Dependency[] and api:org.gradle.api.artifacts.Configuration[] classes.
+You may find that other dependencies will want to pull in the same transitive dependency that misses the artifacts. Alternatively, you can exclude the transitive dependencies for a particular configuration by calling the method api:org.gradle.api.artifacts.Configuration#exclude(java.util.Map)[].
 
-Not every transitive dependency can be excluded - some transitive dependencies might be essential for correct runtime behavior of the application. Generally, one can exclude transitive dependencies that are either not required by runtime or that are guaranteed to be available on the target environment/platform.
+++++
+<sample id="exclude-transitive-for-configuration" dir="userguide/dependencies/excludingTransitiveDependenciesForConfiguration" title="Excluding transitive dependency for a particular configuration">
+    <sourcefile file="build.gradle" snippet="exclude-transitive-dependencies"/>
+</sample>
+++++
 
-Should you exclude per-dependency or per-configuration? It turns out that in the majority of cases you want to use the per-configuration exclusion. Here are some typical reasons why one might want to exclude a transitive dependency. Bear in mind that for some of these use cases there are better solutions than exclusions!
+[NOTE]
+====
+As a build script author you often times know that you want to exclude a dependency for all configurations available in the project. The Groovy DSL uses meta programming capabilities to allow the use of the `all` keyword to define a global rule e.g. `configurations.all { ... }`.
```

Comment:
I specifically mentioned Groovy here because I am not sure if we have the same handling for Kotlin. @bamboo @eskatos Is this supported by the Kotlin DSL and if yes how does it look like?

Reply:
`configurations.all {}` is no magic, see [DomainObjectCollection#all](https://github.com/gradle/gradle/blob/b0032a8a355379e5868b22d660f8be5cf857168e/subprojects/core-api/src/main/java/org/gradle/api/DomainObjectCollection.java#L126).
It is just a regular method invocation on both the Groovy and Kotlin DSLs.

Reply:
Thanks for clarifying. For some reason I thought that it was based on metaprogramming. Will change the text here.